ci/scripts/fetch_data.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+# SPDX-FileCopyrightText: Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import curses
+import logging
+import os
+import subprocess
+import sys
+import tempfile
+import time
+
+LFS_DATASETS = {
+    'all': '**',
+    'examples': 'examples/**',
+    'models': 'models/**',
+    'tests': 'tests/**',
+    'validation': 'models/datasets/validation-data/**'
+}
+
+
+def print_line(stdscr, max_x, last_print_len, line):
+    print_len = min(len(line), max_x)
+    if stdscr is not None:
+        if print_len < last_print_len:
+            stdscr.move(0, 0)
+            stdscr.clrtoeol()
+
+        stdscr.addstr(0, 0, line, print_len)
+        stdscr.refresh()
+    else:
+        logging.info(line)
+
+    return print_len
+
+
+def lfsPull(stdscr, include_paths, poll_interval=0.1):
+    """
+    Performs a git lfs pull.
+    This can take upwards of a minute to complete we will make use of the
+    GIT_LFS_PROGRESS hook to send progress to a file which we can then tail
+    while the command is executing.
+    """
+    cmd = 'git lfs pull -I "{}"'.format(','.join(include_paths))
+    with tempfile.NamedTemporaryFile() as progress_file:
+        env = os.environ.copy()
+        env['GIT_LFS_PROGRESS'] = progress_file.name
```

Comment:
Have you also looked into trying `GIT_LFS_FORCE_PROGRESS`? This might be an easier way to force the progress output without needing curses.